packages/narwallets/src/lib/narwallets.ts

```diff
@@ -0,0 +1,371 @@
+        removePendingPromise(pendingPromise);
+        if (pendingPromise.timeout) {
+          clearTimeout(pendingPromise.timeout);
+        }
+        if (event.data.error) {
+          pendingPromise.reject(event.data.error);
+        } else {
+          pendingPromise.resolve(event.data.result.data);
+        }
+      }
+    }
+  });
+  // let wallet = window.narwallets!;
+
+  // return {
+  //   wallet,
+  // };
+};
+
+const Narwallets: WalletBehaviourFactory<InjectedWallet> = async ({
+  metadata,
+  store,
+  logger,
+}) => {
+  // const _state = setupNarwalletsState();
+
+  const cleanup = () => {
+    // for (const key in _state.wallet.callbacks) {
+    //   _state.wallet.remove(key);
+    // }
+  };
+
+  const signOut = async () => {
+    if (!(await isSignedIn())) {
+      return;
+    }
+
+    cleanup();
+
+    const res = await sendToNarwallets("sign-out");
+    // const res = await _state.wallet.signOut();
+    if (res === true) {
+      return;
+    }
+
+    const error = new Error(
+      typeof res.error === "string" ? res.error : res.error.type
+    );
+
+    // Prevent signing out by throwing.
+    if (error.message === "User reject") {
+      throw error;
+    }
+
+    // Continue signing out but log the issue.
+    logger.log("Failed to sign out");
+    logger.error(error);
+  };
+
+  const setupEvents = () => {
+    // _state.wallet.on("accountChanged", async (newAccountId) => {
+    //   logger.log("onAccountChange", newAccountId);
+    //   emitter.emit("signedOut", null);
+    // });
+    // _state.wallet.on("rpcChanged", async (rpc) => {
+    //   logger.log("onNetworkChange", rpc);
+    //   if (options.network.networkId !== rpc.networkId) {
+    //     await signOut();
+    //     emitter.emit("signedOut", null);
+    //     emitter.emit("networkChanged", { networkId: rpc.networkId });
+    //   }
+    // });
+  };
+
+  // const getAccountId = () => {
+  //   const accountId = _state.wallet.getAccountId();
+
+  //   if (!accountId) {
+  //     return [];
+  //   }
+
+  //   return [{ accountId }];
+  // };
+
+  const isValidActions = (
+    actions: Array<Action>
+  ): actions is Array<FunctionCallAction> => {
+    return actions.every((x) => x.type === "FunctionCall");
+  };
+
+  const transformActions = (actions: Array<Action>) => {
+    const validActions = isValidActions(actions);
+
+    if (!validActions) {
+      throw new Error(
+        `Only 'FunctionCall' actions types are supported by ${metadata.name}`
+      );
+    }
+
+    return actions.map((x) => x.params);
+  };
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ): Array<SignAndSendTransactionParams> => {
+    return transactions.map((transaction) => {
+      return {
+        receiverId: transaction.receiverId,
+        actions: transformActions(transaction.actions),
+      };
+    });
+  };
+
+  if (await isSignedIn()) {
+    setupEvents();
+  }
+
+  return {
```

Comment:
Can you please add a new method called `verifyOwner` in the return here?
If your wallet does not support it you can throw an error as it's done here:
https://github.com/near/wallet-selector/blob/main/packages/ledger/src/lib/ledger.ts#L223-L227